apps/examples/power/power_main.c

```diff
@@ -26,127 +26,172 @@
- */
-static pthread_addr_t power_daemon()
-{
-	int ret = 0;
-	int fd_lock, fd_unlock;
-
-	/* Release PM lock after bootup*/
-	fd_unlock = open(PM_UNLOCK_PATH, O_WRONLY);
-	if (fd_unlock < 0) {
-		printf("Failed to open PM_UNLOCK file, ret: %d\n", fd_unlock);
-		return NULL;
+static int is_running;
+
+
+static int pm_sleep_test(void)  {
+	printf("pm sleep thread start\n");
+	int fd = open(PM_DRVPATH, O_WRONLY);
+	if (fd < 0) {
+		printf("Fail to open pm sleep(errno %d)", get_errno());
+		return -1;
 	}
-	fd_lock = open(PM_LOCK_PATH, O_WRONLY);
-	if (fd_lock < 0) {
-		printf("Failed to open PM_LOCK file, ret: %d\n", fd_lock);
-		close(fd_unlock);
-		return NULL;
+
+	while (is_running) {
+		if (ioctl(fd, PMIOC_SLEEP, 100) < 0) {
+			printf("Fail to pm sleep(errno %d)\n", get_errno());
+			return -1;
+		}
 	}
+	close(fd);
+	return 0;
+}
 
-	ret = write(fd_unlock, NULL, PM_WAKEUP_TIMER_DURATION);
-	if (ret < 0) {
-		printf("Failed to unlock PM state transition, ret: %d\n", ret);
-		goto errout_with_fd;
+static int pm_suspend_resume_test(void)
+{
+	printf("pm supend resume repeat test start\n");
+	pm_domain_arg_t domain_arg;
+	int domain_id;
+	int test_count = 0;
+	int	fd = open(PM_DRVPATH, O_WRONLY);
+	if (fd < 0) {
+		printf("Fail to open pm sleep(errno %d)", get_errno());
+		return -1;
 	}
-	printf("Boot completed ..Unlocked PM!!\n");
-
-	do {
-		printf("Locking PM state, transition not expected during APP sleep\n");
-		/* Apply PM lock to prevent sleep*/
-		ret = write(fd_lock, NULL, 0);
-		if (ret < 0) {
-			printf("Failed to lock PM state transition, ret: %d\n", ret);
-			goto errout_with_fd;
-		}
 
-		/* Trigger APP sleep, but we do not expect PM state transition */
-		sleep(PM_DAEMON_SLEEP_INTERVAL);
-		printf("1st APP sleep complete\n\n");
+	domain_arg.domain_name = "SUSPEND_TEST";
+	if (ioctl(fd, PMIOC_DOMAIN_REGISTER, &domain_arg) < 0) {
+		printf("Fail to register pm domain(errno %d)", get_errno());
+		close(fd);
+		return -1;
+	}
+	domain_id = domain_arg.domain_id;
+
+	printf("%d domain id\n", domain_id);
 
-		printf("Unlocking PM state, transition expected during APP sleep\n");
-		/* Release PM lock for sleep */
-		ret = write(fd_unlock, NULL, PM_WAKEUP_TIMER_DURATION);
-		if (ret < 0) {
-			printf("Failed to unlock PM state transition, ret: %d\n", ret);
-			goto errout_with_fd;
+	while (is_running) {
+		test_count++;
+		printf("call suspend (%d)\n", test_count);
+		if(ioctl(fd, PMIOC_SUSPEND, domain_id) < 0) {
+			printf("Fail to suspend(errno %d)\n", get_errno());
+			break;
 		}
 
-		/* Trigger APP sleep, this time we expect PM state transition */
-		sleep(PM_DAEMON_SLEEP_INTERVAL);
-		printf("2nd APP sleep complete\n\n");
+		sleep(2);
 
-		printf("##########################################################\n\n");
-	} while (1);
+		printf("call resume(%d)\n", test_count);
+		if(ioctl(fd, PMIOC_RESUME, domain_id) < 0) {
+			printf("Fail to resume(errno %d)\n", get_errno());
+			break;
+		}
+
+		sleep(18);
```

Comment:
Any reason to sleep(18)? It seems better to define a macro for duration instead of hardcoding it.